presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp

```diff
@@ -643,4 +660,48 @@ void PeriodicTaskManager::addHttpEndpointLatencyStatsTask() {
       kHttpEndpointLatencyPeriodGlobalCounters,
       "http_endpoint_counters");
 }
+
+void PeriodicTaskManager::addWatchdogTask() {
+  addTask(
+      [this,
+       deadlockedTasks = std::vector<std::string>(),
+       opCalls = std::vector<velox::exec::Task::OpCallInfo>()]() mutable {
+        deadlockedTasks.clear();
+        opCalls.clear();
+        if (!taskManager_->getLongRunningOpCalls(
+                stuckDriverThresholdMs_, deadlockedTasks, opCalls)) {
+          LOG(ERROR)
+              << "Cannot take lock on task manager, likely starving or deadlocked";
+          RECORD_METRIC_VALUE(kCounterNumTasksDeadlock, 1);
```

Comment:
this fails on task manager lock? Shall we have a different counter? like TaskManagerLockTimeOut?